### PR TITLE
Improve optimized cache

### PR DESCRIPTION
**Pull request by javiercn**
* Remove unnecessary files from the final build.
* Add support for specifying the runtimes we want to generate the cache for.
* Add support for custom prefixes on the zip file.

### CODE REVIEW

**Comment by javiercn** [comment]
@pranavkm

File: src/DependenciesPackager/Program.cs [hunk 1]
```diff
@@ -226,6 +248,23 @@ private int Execute()
...
+        {
+            return runtime.Substring(runtime.LastIndexOf('-') + 1);
+        }
+
+        private void RemoveUnnecesaryHiveFiles(string cacheBasePath, Project project)
+        {
+            var directories = new DirectoryInfo(cacheBasePath);
+            foreach (var directory in directories.EnumerateDirectories())
+            {
+                if (!project.Dependencies.Any(d => d.Name.Equals(directory.Name)))
```
Comment (pranavkm) [inline src/DependenciesPackager/Program.cs]: casing?

File: src/DependenciesPackager/Program.cs [hunk 2]
```diff
@@ -234,14 +273,15 @@ private void RemoveUnnecessaryRestoredFiles()
             var dlls = Directory.GetFiles(restorePath, "*.dll", SearchOption.AllDirectories);
```
Comment (pranavkm) [inline src/DependenciesPackager/Program.cs]: Could you add these to a list? 

```
var patternToExclude = new[] { ".dll", "crossgen.exe", "clrjit.dll" };
var files = patternsToExclude.SelectMany(..);
```

**Comment by pranavkm** [comment]
:shipit:

**Pull request closed without merging**